pkg/solana/chainreader/bindings_test.go

```diff
@@ -104,3 +107,121 @@ func (_m *mockBinding) QueryKey(
 
 	return ret.Get(0).([]types.Sequence), ret.Error(1)
 }
+
+func Test_namespaceBinding_BindReaders(t *testing.T) {
+	type fields struct {
+		name    string
+		mu      sync.RWMutex
+		readers map[string]readBinding
+		bound   map[solana.PublicKey]bool
+	}
+	type args struct {
+		ctx     context.Context
+		address solana.PublicKey
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "no readers",
+			fields: fields{
+				name:    "testNamespace",
+				readers: make(map[string]readBinding),
+				bound:   make(map[solana.PublicKey]bool),
+			},
+			args: args{
+				ctx:     context.Background(),
+				address: solana.PublicKey{},
+			},
+			wantErr: false,
+		},
+		{
+			name: "single reader binds successfully",
+			fields: fields{
+				name: "testNamespace",
+				readers: map[string]readBinding{
+					"reader1": func() *mockBinding {
+						m := &mockBinding{}
+						m.On("Bind", mock.Anything, mock.Anything).Return(nil)
+						return m
+					}(),
+				},
+				bound: make(map[solana.PublicKey]bool),
+			},
+			args: args{
+				ctx:     context.Background(),
+				address: solana.PublicKey{},
+			},
+			wantErr: false,
+		},
+		{
+			name: "multiple readers bind successfully",
+			fields: fields{
+				name: "testNamespace",
+				readers: map[string]readBinding{
+					"reader1": func() *mockBinding {
+						m := &mockBinding{}
+						m.On("Bind", mock.Anything, mock.Anything).Return(nil)
+						return m
+					}(),
+					"reader2": func() *mockBinding {
+						m := &mockBinding{}
+						m.On("Bind", mock.Anything, mock.Anything).Return(nil)
+						return m
+					}(),
+				},
+				bound: make(map[solana.PublicKey]bool),
+			},
+			args: args{
+				ctx:     context.Background(),
+				address: solana.PublicKey{},
+			},
+			wantErr: false,
+		},
+		{
+			name: "reader bind returns error",
+			fields: fields{
+				name: "testNamespace",
+				readers: map[string]readBinding{
+					"reader1": func() *mockBinding {
+						m := &mockBinding{}
+						// Returns nil to show the first reader binds successfully
+						m.On("Bind", mock.Anything, mock.Anything).Return(nil)
+						return m
+					}(),
+					"reader2": func() *mockBinding {
+						m := &mockBinding{}
+						// Returns an error to simulate a bind failure
+						m.On("Bind", mock.Anything, mock.Anything).Return(fmt.Errorf("bind error"))
+						return m
+					}(),
+				},
+				bound: make(map[solana.PublicKey]bool),
+			},
+			args: args{
+				ctx:     context.Background(),
+				address: solana.PublicKey{},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &namespaceBinding{
+				name:    tt.fields.name,
+				mu:      tt.fields.mu,
+				readers: tt.fields.readers,
+				bound:   tt.fields.bound,
+			}
+			err := b.BindReaders(tt.args.ctx, tt.args.address)
```

Comment:
nit: Couldn't we just initialize a context and just pass it in directly here? Also I think we prefer using `tests.Context(t)` for test contexts. That way we can just replace the args struct altogether and pass the address directly for each test above.
```suggestion
			err := b.BindReaders(tests.Context(t), tt.args.address)
```

Reply:
good call cleaned up a bit